File: Codeforces/A/151A-Soft_Drinking.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long int ll;

int main()
{
    int n, k, l, c, d, p, nl, np;
    cin >> n >> k >> l >> c >> d >> p >> nl >> np;

    int drink = (k*l)/nl;
    int lime = c*d;
    int salt = p/np;
    int res = min(drink, min(lime, salt))/n;
    cout << res << endl;
    return 0;
}

File: Codeforces/A/151A-Soft_Drinking.go
package main

import (
	"fmt"
	"math"
)

func main() {

	var n, k, l, c, d, p, nl, np int
	fmt.Scan(&n, &k, &l, &c, &d, &p, &nl, &np)

	drink := (k*l)/nl
	lime := c*d
	salt := p/np

	res:= math.Min(float64(drink),math.Min(float64(lime),float64(salt)))
	fmt.Println(int(res)/n);

}